pkg/util/testing/wrapper.go

```diff
@@ -129,6 +144,50 @@ func (j *JobSetWrapper) NumNodes(numNodes int32) *JobSetWrapper {
 	return j
 }
 
+func (j *JobSetWrapper) Parallelism(rName string, p int32) *JobSetWrapper {
+	for i, rJob := range j.Spec.ReplicatedJobs {
+		if rJob.Name == rName {
+			j.Spec.ReplicatedJobs[i].Template.Spec.Parallelism = &p
+		}
+	}
+	return j
+}
+
+func (j *JobSetWrapper) Completions(rName string, c int32) *JobSetWrapper {
+	for i, rJob := range j.Spec.ReplicatedJobs {
+		if rJob.Name == rName {
+			j.Spec.ReplicatedJobs[i].Template.Spec.Completions = &c
+		}
+	}
+	return j
+}
+
+func (j *JobSetWrapper) LauncherReplica() *JobSetWrapper {
+	for i, rJob := range j.Spec.ReplicatedJobs {
+		if rJob.Name == constants.JobTrainerNode {
```

Comment:
Do we really need this check ?
E.g. JobSet wrapper always has Trainer Node Job: https://github.com/kubeflow/trainer/blob/76dff03117d3a191b2c625b116344f980d389686/pkg/util/testing/wrapper.go#L89

Reply:
This is needed for an array right shift operation.
ReplicatedJob must have Jobs in the following order:

- Initializer
- Laucher
- Trainer

Reply:
@tenzen-y If we don't really use DependsOn API for Launcher <-> Trainer relationship, do we need to preserve order of ReplicatedJob (Launcher is set before Trainer Node) in our testing ?

Reply:
Slice should be guaranteed the order. So, we want to keep using same order, anyplaces.